script/ensure-node-source-fixture-files.js

```diff
@@ -0,0 +1,155 @@
+      .then(async function () {
+        console.log(`Successfully downloaded file from ${details.url}.`)
+        logVerbose('checking if hash matches in for...of loop.');
+        // Check its hash
+        const hashDidMatch = await verifyHash(destinationPath, details.sha256sum);
+        if (!hashDidMatch) {
+          console.error(`Hash did not match for ${destinationPath}`);
+        }
+      })
+      .catch(console.error);
+  }
+}
+
+async function verifyExistingFile (targetPath, expectedHash) {
+  if (fs.existsSync(targetPath)) {
+    logVerbose(targetPath + " already exists.");
+
+    logVerbose(`verifying hash for ${targetPath} in verifyExistingFile`)
+    const hashDidMatch = await verifyHash(targetPath, expectedHash)
+
+    if (hashDidMatch) {
+      // Successfully verified existing file. Return true.
+      console.log(`Existing file ${targetPath} successfully verified.`)
+      return true;
+    } else {
+      // Existing file's hash was wrong. Delete the old file.
+      console.error(`Hash did not match for ${targetPath}. Deleting.`);
+      fs.rmSync(targetPath);
+    }
+  } else {
+    // File did not actually exist.
+    logVerbose(targetPath + " does not exist.");
+  }
+  // If we haven't returned yet, verification did not succeed. Return false.
+  return false;
+}
+
+async function verifyHash (path, expectedHash) {
+  // Checks the hash of a given locally downloaded file against its expected value.
+  // Returns true/false.
+  // See: https://nodejs.org/api/crypto.html#class-hash for details.
+
+  // This module apparently needs to be required each time this function runs,
+  // or hashing multiple files in parallel can get messed up? Not sure why.
+  const { createHash } = require('node:crypto');
+
+  const hash = createHash('sha256');
+  hash.update(fs.readFileSync(path))
+  const actualHash = hash.digest('hex');
+  hash.end()
+  // The hash of the actual file on disk.
+
+  logVerbose("expectedHash is: " + expectedHash);
+  logVerbose("actualHash is: " + actualHash);
+  // The expected hash is from the array of objects at the top of the file.
+
+  if (actualHash === expectedHash) {
+    logVerbose('Hash verified successfully');
+    return true;
+  } else {
+    logVerbose('Hash did not match');
+    return false;
+  }
+}
+
+function downloadFileToDestination(url, filePath, expectedHash) {
+  // Actually downloads the desired file to disk.
+  // Returns a Promise.
+  // Based on https://scrapingant.com/blog/download-image-javascript
+
+  logVerbose(`Downloading ${url} to ${filePath}`);
+
+  return new Promise((resolve, reject) => {
+    https.get(url, (res) => {
+      // Technically any 2XX series HTTP response code means success,
+      // but in practice this should always be exactly "200"? Adjust if needed.
+      if (res.statusCode === 200) {
+        res.pipe(fs.createWriteStream(filePath))
+          .on('error', reject)
+          .once('close', resolve);
```

Comment:
```suggestion
          .on("error", reject)
          .once("close", resolve);
```
Codacy suggestion: Single quote -> Double quote